src/components/Shared/GlobalIntegrationGrid/index.js

```diff
@@ -0,0 +1,178 @@
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import { useResourceQuery } from 'src/hooks/resource'
+
+const GlobalIntegrationGrid = ({ masterSource, masterId}) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const {
+    labels: _labels,
+    access,
+  } = useResourceQuery({
+    datasetId: ResourceIds.IntegrationAccount,
+  })
+
+  const { formik } = useForm({
+    maxAccess: access,
+    enableReinitialize: false,
+    validateOnChange: true,
+    initialValues: {
+      Integrations: [
+        {
+          id: 1,
+          postTypeId: null,
+          ptName: '',
+          accountId: null,
+          accountRef: null,
+          accountName: ''
+        }
+      ]
+    },
+    onSubmit: values => {
+      postIntegrations(values);
+    }
+  });
+  
+  const postIntegrations = values => {
+    
+    const filteredIntegrations = values.Integrations.filter(integration => integration.accountId !== null);
+
+    const data = { 
+      masterSource: masterSource,
+      masterId: masterId,
+      integrationAccounts: filteredIntegrations
+    };
+
+    postRequest({
+      extension: GeneralLedgerRepository.IntegrationAccounts.set2,
+      record: JSON.stringify(data)
+    })
+    .then(res => {
+      toast.success('Record Successfully Saved');
+    })
+    .catch(error => { });
+  };
+  
+    const column = [
+      {
+        component: 'textfield',
+        label: _labels.ptName,
+        name: 'ptName',
+        props:{readOnly: true}
+      },
+      {
+        component: 'resourcecombobox',
+        label: _labels.account,
+        name: 'accountRef',
+        props: {
+          endpointId: GeneralLedgerRepository.IntegrationAccounts.qry,
+          valueField: 'accountId',
+          displayField: 'accountRef',
+          mapping: [ { from: 'accountName', to: 'accountName' } ,
+          { from: 'accountRef', to: 'accountRef' },
+          { from: 'accountId', to: 'accountId' } ],
+          columnsInDropDown: [
+            { key: 'accountRef', value: 'accountRef' },
+            { key: 'accountName', value: 'accountName' },
+          ]
+        }
+      },
+      {
+        component: 'textfield',
+        label: _labels.accountName,
+        name: 'accountName',
+        props:{readOnly: true}
+      }
+    ]
+
+    useEffect(() => {
+      getPostTypes();
+    }, []); 
+
+    const getPostTypes = () => {
+      getRequest({
+        extension: GeneralLedgerRepository.IntegrationPostTypes.qry
+      })
+      .then(res => {
+        if (res.list.length > 0) {
+          const postTypes = res.list.map((record, index) => ({
```

Comment:
you are looping twice on the post types list
once here and once after getting the integration accounts list in getIntegrations( ).
You can loop only once for a better performance.